### ISSUE TITLE
MeshInstance2D (with ArrayMesh) disappears randomly when camera moves past a certain X / Y coordinate

### ISSUE THREAD

**Issue by Beider** [godotengine/godot#59518]
### Godot version

3.4.4 stable mono

### System information

Windows 10, GLES2 / GLES3, GeForce RTX 2080 TI, 511.79

### Issue description

I am trying to generate a terrain using MeshInstance2D instead of a tilemap. I generate the terrain in 8x8 chunks and I use AddSurfaceFromArrays to add the surface data to my ArrayMesh.

I already added a separate issue regarding some lag I am experiencing [here](https://github.com/godotengine/godot/issues/59517).

Another problem is that suddenly the mesh just disappears, I observe the following symptoms:

- Always happens at some X or Y coordinate (or both).
- Sometimes the different MeshInstance2D disappear at slightly different coordinates but they always do disappear
- Resizing godot (double clicking window title) makes them come back, even if I resize back to the original size.

[Here is a video](https://www.youtube.com/watch?v=bdGf67Fx2Ds&ab_channel=VegardBeider) showing the effect in action, the second part of the video shows the meshes disappearing at the same time. I have tried to disable 2D batching in the project and also tried with both GLES2 and GLES3. 

### Steps to reproduce

I attached a project, you just need to run it with 3.4.4 mono and start it. Instructions are in the label inside the project.

### Minimal reproduction project

[MeshInstance2DProblem.zip](https://github.com/godotengine/godot/files/8351864/MeshInstance2DProblem.zip)

**Comment by Calinou** [comment]
cc @lawnjelly 

I'd recommend creating a minimal reproduction project that uses only GDScript to make reproduction easier.

**Comment by lawnjelly** [comment]
Yes probably needs a gdscript MRP.

It may just be that you need to set a custom AABB at a guess, I'm not super familiar with how the AABB is meant to be created when you generate a mesh manually, whether it is meant to be created automatically or it is your job.

If the AABB is garbage the culling won't work as expected.

**Comment by Beider** [comment]
> Yes probably needs a gdscript MRP.
> 
> It may just be that you need to set a custom AABB at a guess, I'm not super familiar with how the AABB is meant to be created when you generate a mesh manually, whether it is meant to be created automatically or it is your job.
> 
> If the AABB is garbage the culling won't work as expected.

I wasn't aware AABB was even a thing so this was my bad, I set an AABB now like this,

`((ArrayMesh)instance.Mesh).CustomAabb = new AABB(new Vector3(float.MinValue, float.MinValue, float.MinValue), new Vector3(float.MaxValue, float.MaxValue, float.MaxValue));`

And it seems to work now, I tried 4 times and the problem of meshes disappearing was gone. That being said, perhaps this is something that should get some more attention in the documentation. I realise my usecase is probably different than most, but dynamic meshes probably is not too unusual. 

I would propose mentioning this on either (or all) of:
https://docs.godotengine.org/en/stable/classes/class_meshdatatool.html
https://docs.godotengine.org/en/3.1/tutorials/content/procedural_geometry.html
https://docs.godotengine.org/en/3.1/classes/class_surfacetool.html

I did read all of these docs before starting with dynamic mesh generation but this was not mentioned explicitly anywhere, just a short note that you need to consider the AABB if you generate a large mesh would be nice.

@Calinou Does it sound like a good idea to add this to the docs? If so should I file another issue for this?
This can be closed now as the AABB was the problem.

**Comment by Calinou** [comment]
I thought the AABB was supposed to be automatically generated – I think this is the case in 3D at least. If that doesn't work in 2D, this is surely a bug.

**Comment by lawnjelly** [comment]
Well it would still be nice to have an MRP in gdscript to see what is happening, and try and work out from the Godot code whether it is trying to create an AABB (and failing) or just it is left to the user.

Often with dynamic meshes (not just in Godot) it is left to the user to supply an AABB - because generating the AABB can be relatively expensive, and in many cases the user will provide an "oversize" AABB that accounts for any movement. The same problem happens particularly with skinning.
